src/window/renderer.rs

```diff
@@ -13,24 +15,25 @@ fn create_surface(
     gr_context: &mut DirectContext,
     fb_info: FramebufferInfo,
 ) -> Surface {
-    let pixel_format = windowed_context.get_pixel_format();
+    let pixel_format = windowed_context.get_config();
     let size = windowed_context.window().inner_size();
     let size = (
         size.width.try_into().expect("Could not convert width"),
         size.height.try_into().expect("Could not convert height"),
     );
     let backend_render_target = BackendRenderTarget::new_gl(
         size,
+        Some(pixel_format.num_samples() as usize),
         pixel_format
-            .multisampling
-            .map(|s| s.try_into().expect("Could not convert multisampling")),
-        pixel_format
-            .stencil_bits
+            .stencil_size()
             .try_into()
             .expect("Could not convert stencil"),
         fb_info,
     );
-    windowed_context.resize(size.into());
+    windowed_context.resize(
+        NonZeroU32::new(size.0 as u32).unwrap(),
+        NonZeroU32::new(size.1 as u32).unwrap(),
```

Comment:
I think we should force a minimum size here, rather than crashing with unwrap errors. I will fix that myself.

Actually I will also add a maximum size of 8k or something, if someone has a bigger window than that, it's better to get a blurry one, than to crash because of too big buffers. I think we have some reports for that as well. I will add the same limit to the creation as well.

Reply:
I added validation for all the buffer sizes.